sds_data_manager/lambda_code/SDSCode/api_lambdas/utils.py

```diff
@@ -0,0 +1,30 @@
+"""API utils."""
+
+import logging
+
+logger = logging.getLogger()
+logger.setLevel(logging.INFO)
+
+
+def is_authenticated_user(event):
+    """Check if the API path is authenticated, allowing access to unreleased files.
+
+    This function examines the routeKey and rawPath in the event to determine
+    if the request is coming through an authenticated path (containing 'api-key'
+    or 'auth'). Authenticated paths have access to all files, while
+    non-authenticated paths only have access to released files.
+
+    Parameters
+    ----------
+    event : dict
+        The API Gateway event object
+
+    Returns
+    -------
+    bool
+        True if the path is authenticated, False otherwise
+    """
+    rk = event.get("routeKey", "")
+    route_key = rk.split("/")[1] if "/" in rk else ""
+    logger.error(f"Route Key: {route_key}")
+    return route_key in ("authorized", "api-key")
```

Comment:
I find this easier to read than a split personally.

The logger.error should be changed to a different log level, but I'd suggest just removing it altogether because this will be called a lot I think.
```suggestion
    return event.get("rawPath", "").startswith(("/authorized", "/api-key"))
```

Reply:
Sounds good. But I tried that starts with but note that the routeKey doesn't start with /authorized but with something like this 'GET /authorized'. That's why I still had to split by '/'.

Reply:
I switched it to "rawPath" which does I think.